org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ProjectUtils.java

```diff
@@ -235,6 +235,72 @@ public static boolean isOnSourcePath(IPath sourcePath, IJavaProject project) thr
 		return false;
 	}
 
+	public static boolean setInvisibleProjectOutputPath(IJavaProject javaProject, String outputPath, IProgressMonitor monitor) throws CoreException {
```

Comment:
This utility is just used for invisible project only, it's better to move to InvisibleProjectImporter.java.